pruner/src/pruner_util.cpp

```diff
@@ -262,13 +261,21 @@ int check_pruned_program(const IR::P4Program **orig_program,
         INFO("File has not changed. Skipping analysis.");
         return EXIT_SUCCESS;
     }
-    int exit_code = get_exit_info(out_file, pruner_conf).exit_code;
+    ExitInfo exit_info = get_exit_info(out_file, pruner_conf);
+    int exit_code = exit_info.exit_code;
     // if got the right exit code, then modify the original program, if not
     // then choose a smaller bank of statements to remove now.
     if (exit_code != pruner_conf.exit_code) {
         INFO("FAILED");
         return EXIT_FAILURE;
     } else {
+        // When checking crash bugs, there must be the string Compiler Bug in the 
+        // output (the exit code can remain the same if there is another error)
+        if(pruner_conf.err_type == ErrorType::CrashBug && exit_info.err_msg.find("Compiler Bug") == NULL){
```

Comment:
CrashBug should already include this check, shouldn't it? This is how defined it. If the classification is wrong, we need to fix it earlier.

Reply:
We usually have the enum label Error for this and  it should be classified as you mentioned. Are we not using it?

Reply:
Right but we only call classify_bug initially when we receive the original program, but during our pruning process we never call classify_bug, so what i'm saying is lets say we have two versions of the program before_prune_iteration1 and after_prune_iteration1, and suppose the prune iteration removed some statements from the program which caused a syntax error, so in this case we want this iteration to "fail" right instead of "pass", but in our earlier implementation we assumed that removing statements can't lead to a syntax error i think.

Reply:
We probably should reclassify every time then, to make sure that we do not lose the compiler bug. I think there was a reason why we did it differently but we should review that.